atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/NumberOfShardsProvider.java

```diff
@@ -43,16 +43,10 @@ default int getNumberOfShards(SweeperStrategy strategy) {
     private static int getNumberOfShards(
             ShardProgress progress, Supplier<Integer> shards, MismatchBehaviour mismatchBehaviour) {
         switch (mismatchBehaviour) {
-            case THROW:
-                int shardsFromSupplier = shards.get();
-                int shardsFromProgress = progress.getNumberOfShards();
-                if (shardsFromSupplier != shardsFromProgress) {
-                    throw new SafeIllegalStateException(
-                            "Number of shards from supplier and progress do not match",
-                            SafeArg.of("shardsFromSupplier", shardsFromSupplier),
-                            SafeArg.of("shardsFromProgress", shardsFromProgress));
-                }
-                return shardsFromSupplier;
+            case IGNORE_UPDATES:
+                // Here, we will never update the number of shards from config. This is because the ASTS work requires
+                // knowing the number of shards at all time from the beginning of install, and changing is quite painful
+                return progress.getNumberOfShards();
```

Comment:
Discussed offline: in particular, with throwing, if the user had some number of shards < 128, bootstrapping would set that to 128, and then there would thereafter be a mismatch unless the user actually changes the config, which is unreasonable.